File: [프로그래머스] 코딩테스트 고득점 Kit/정렬/Level 1/최소직사각형.java
// https://school.programmers.co.kr/learn/courses/30/lessons/86491

import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int max_width = 0, max_height = 0;

        for(int i = 0; i < sizes.length; i++) {
            Arrays.sort(sizes[i]);
            max_width = Math.max(max_width, sizes[i][0]);
            max_height = Math.max(max_height, sizes[i][1]);
        }

        return max_width * max_height;
    }
}

File: [프로그래머스] 코딩테스트 고득점 Kit/완전탐색/Level 1/최소직사각형.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/86491

#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int max_width = 0, max_height = 0;

    for (auto size : sizes) {
        // 가로, 세로 중 큰 값이 가로가 되도록 정렬
        sort(size.begin(), size.end());
        max_width = max(max_width, size[0]);
        max_height = max(max_height, size[1]);
    }

    return max_width * max_height;
}